train_heads_selection.py

```diff
@@ -89,6 +94,7 @@ def compute_loss(vocabs: Vocabs, mask: torch.Tensor,
   pad_idx = vocabs.relation_vocab[PAD]
   binary_outputs = (gold_outputs != no_rel_index) * (gold_outputs != pad_idx)
   binary_outputs = binary_outputs.type(torch.FloatTensor)
+  mask = create_mask(gold_outputs, concepts_lengths, config) if mask is None else mask
```

Comment:
if mask is None shouldn't you rather not mask the loss? I would assume that would make more sense

Reply:
No, so here I thought about it like this: this is done in compute_loss and it needs a mask. Good, but we also need a mask for computing the coloring scores on the image. So because the mask computing has some randomness in it, I wanted to make sure that it was the EXACT SAME mask! And that is why it is given as a parameter.
Now why the 'if mask is None' you may wonder, well initially this was done ONLY for evaluation as only the eval flow had images, but later on I added them also on the train flow. (So basically the if mask is None is now useless as it will never be None).
The idea was that if the mask was None, it was not computed yet, so let's compute it now, and acutally I left that if there just for safety, but TBH now that I think about it I will remove it as it does not make sense anymore.